api/drivers/aws/driver.js

```diff
@@ -255,6 +295,67 @@ class AWSDriver extends Driver {
       });
     });
   }
+
+  /**
+   * Calculate credit used by a user
+   *
+   * @method getUserCredit
+   * @param {user} user User to calculate credit usage from
+   * @return {Promise[number]}
+   */
+  getUserCredit(user) {
+    let duration = [];
+    let product = [];
+    let priceAll = [];
+    var finalPrice = 0;
+    var jsonPrice = {};
+
+    return new Promise((resolve, reject) => {
+      Promise.props({
+        conf: ConfigService.get('price'),
+        timeHistory: UserService.getUserHistory(user, 'aws')
+      })
+        .then((result) => {
+          jsonPrice = result.conf.price;
+          duration = result.timeHistory;
+          var prod = Object.keys(jsonPrice.products);
+          return duration.forEach((element) => {
+            return prod.forEach((elem) => {
+              var obj = jsonPrice.products[elem].attributes;
+              if (obj.instanceType === element.type) {
+                product.push(jsonPrice.products[elem]);
+                return jsonPrice.products[elem];
+              } else {
+                return null;
+              }
+            });
+          });
+        })
+        .then(() => {
+          return product.forEach((element) => {
+            var tmp = {};
+            var tmpKey = null;
+
+            tmp = jsonPrice.terms.OnDemand[element.sku];
+            tmpKey = Object.keys(tmp);
+            tmp = tmp[tmpKey].priceDimensions;
+            tmpKey = Object.keys(tmp);
+            priceAll.push(tmp[tmpKey].pricePerUnit.USD);
+          });
+        })
+        .then(() => {
+          return duration.forEach((element, index) => {
+            finalPrice += (element.time * priceAll[index]);
+          });
+        })
+        .then(() => {
```

Comment:
You should use `.return(finalPrice)` or better, retrieve the value from the previous Promise.